### PR TITLE
Add docs for truthy/falsy values

### PR DESCRIPTION
**Pull request by nielsbom**
In a number of places in the docs (`!/`, `&&/2`, `Enum.all?`) the
concept of a truthy/falsy value is mentioned and briefly explained.
I propose to have single place for that explanation and point to it
from those other places.

I'm not sure if this is the right location though :-)

### CODE REVIEW

**Comment by whatyouhide** [comment]
Seems like tests are failing because of the formatter. Can you run `mix format` on the Elixir codebase and see if anything changes?

**Comment by whatyouhide** [comment]
Thanks @nielsbom!💟

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -173,6 +173,42 @@ defmodule Kernel do
...
   Those functions will be explicitly marked in their docs as
   "inlined by the compiler".
+
+  ## Truthy and falsy values
+
+  Besides the booleans `true` and `false` Elixir also has the
+  concept of a "truthy" or "falsy" value.
+
+    *  a value is truthy when it is neither `false` nor `nil`
+    *  a value is falsy when it is `false` or `nil`
```
Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]:
```
    *  a value is falsy when it is either `false` or `nil`
```

Comment (nielsbom) [reply lib/elixir/lib/kernel.ex]: Good point => #8859

**Pull request merged**